src/markdown/general/mas_data_voor_vogelatlas.Rmd

```diff
@@ -0,0 +1,254 @@
+        ymd(paste(jaar, r3_start, sep = "-")),
+        ymd(paste(jaar, r3_stop, sep = "-"))) ~ "R3",
+      datum %within% interval(
+        ymd(paste(jaar, r4_start, sep = "-")),
+        ymd(paste(jaar, r4_stop, sep = "-"))) ~ "R4"
+    )) %>%
+  filter(!is.na(periode_in_jaar))
+```
+
+We voegen de coördinaten toe van de telpunten.
+
+```{r}
+steekproef_leemstreek_telpunten <- read_sf(
+  here(data_path, "steekproef_leemstreek_telpunten.geojson")
+  ) %>%
+  mutate(x_plot = unlist(map(.data$geometry, 1)),
+         y_plot = unlist(map(.data$geometry, 2)))
+steekproef_zandleemstreek_telpunten <- read_sf(
+  here(data_path, "steekproef_zandleemstreek_telpunten.geojson")
+  ) %>%
+  mutate(x_plot = unlist(map(.data$geometry, 1)),
+         y_plot = unlist(map(.data$geometry, 2)))
+
+steekproef_mbag <- bind_rows(steekproef_leemstreek_telpunten,
+                             steekproef_zandleemstreek_telpunten) %>%
+  rename(plotnaam = pointid) %>%
+  select(plotnaam, x_plot, y_plot)
+```
+
+
+We herberekenen de afstanden tussen de waarnemingen en de telpunten. We ronden af to 1 m nauwkeurig.
+
+```{r}
+mbag_controle_afstand <- mbag_telperiodes %>%
+  select(oid, plotnaam, distance2plot, naam, jaar, waarnemer = waarneme) %>%
+  inner_join(steekproef_mbag %>% as_tibble(),
+             by = "plotnaam", suffix = c(".stip", ".telpunt")) %>%
+  mutate(calc_distance = st_distance(.$geometry.stip, .$geometry.telpunt,
+            by_element = TRUE),
+         calc_distance = round(units::drop_units(calc_distance))) %>%
+  st_drop_geometry()
+```
+
+We selecteren vervolgens alle waarnemingen **binnen een afstand van 300 m**.
+
+```{r}
+mbag_distance <- mbag_telperiodes %>%
+  full_join(mbag_controle_afstand, 
+            by = join_by(oid, plotnaam, naam, jaar, distance2plot)) %>%
+  select(-c(distance2plot, geometry.telpunt, waarneme)) %>%
+  rename(distance2plot = calc_distance) %>%
+  filter(distance2plot <= 300)
+```
+
+## Berekening maximaal aantal
+
+We berekenen het maximum aantal individuen per jaar per telcirkel voor elke soort.
+
+```{r}
+mbag_atlas <- mbag_distance %>%
+  st_drop_geometry() %>%
+  group_by(plotnaam, jaar, periode_in_jaar, naam) %>%
+  mutate(totaal = sum(aantal)) %>%
+  group_by(plotnaam, jaar, naam, x_plot, y_plot) %>%
+  summarise(max_aantal = max(totaal), .groups = "drop") %>%
+  select(plotnaam, jaar, naam, max_aantal, x_plot, y_plot) %>%
+  mutate(x_coord = x_plot,
+         y_coord = y_plot)
+```
+
+Zet om naar sf object met telcirkels.
+
+```{r}
+mbag_atlas_sf <- st_as_sf(mbag_atlas,
+                          coords = c("x_plot", "y_plot"),
+                          crs = 31370) %>%
+  st_buffer(300)
+```
+
+
+# Output shape files 
+
+We voegen de data samen in een bestand en selecteren jaren 2020-2023.
+
+```{r}
+vogelatlas_out_sf <- bind_rows(mas_piloot_atlas_sf, mbag_atlas_sf) %>%
+  filter(jaar %in% 2020:2023)
+```
+
+We schrijven per jaar een shape file weg.
+
+```{r}
+dir.create(here("data", "vogelatlas"))
```

Comment:
```suggestion
fs::dir_create(here("data", "vogelatlas"))
```

gaat niet protesteren als de map al bestaat

Reply:
Nog een vraag. Bij `sum()` en `max()` heb je nooit `na.rm = TRUE` toegevoegd. Ik veronderstel dat het niet nodig was?

Reply:
Er is ook een argument in die functie om de warning te onderdrukken maar had dat bewust niet gedaan omdat ik het wel een nuttige warning vind.

Nee er zijn nooit NA's. Omdat dit goed is ingevuld of misschien kan je als teller zelfs geen waarnemingen invoeren zonder een aantal aan te geven (?).